File: scripts/start-dev.mjs
import { spawn } from 'child_process';
import path from 'path';

const projectPath = '/vercel/share/v0-project';
const pnpmPath = 'pnpm';

console.log('[v0] Starting Vite dev server...');

const dev = spawn(pnpmPath, ['dev'], {
  cwd: projectPath,
  stdio: 'inherit',
  shell: true
});

dev.on('error', (err) => {
  console.error('[v0] Error starting dev server:', err);
  process.exit(1);
});

dev.on('exit', (code) => {
  console.log('[v0] Dev server exited with code:', code);
});

process.on('SIGINT', () => {
  dev.kill();
  process.exit(0);
});

File: scripts/start_dev.py
#!/usr/bin/env python3
import subprocess
import os
import sys

os.chdir('/vercel/share/v0-project')
print("[v0] Starting Vite development server...")

try:
    result = subprocess.run(['pnpm', 'dev'], check=False)
    sys.exit(result.returncode)
except Exception as e:
    print(f"[v0] Error: {e}")
    sys.exit(1)
